src/electionScraper.js

```diff
@@ -0,0 +1,31 @@
+import Election from "./election";
+
+export const INVALID_ELECTION_ID = -1;
+
+export class ElectionScraper {
+
+    /** @type {Map<number, Election>} */
+    elections = new Map();
+
+    /**
+     * @summary extracts election id from election URL
+     * @param {string} electionUrl election URL to extract from
+     * @returns {number}
+     */
+    static getIdFromElectionUrl(electionUrl) {
+        const [, id] = /^https:\/\/(?:www\.)?(?:[^\/]+\.)+(?:com|net|org)\/election\/(\d+)$/.exec(electionUrl) || [];
+        return id ? +id : INVALID_ELECTION_ID;
+    }
+
+    /**
+     * @summary finds election URLs to scrape
+     */
+    async findElectionUrls() {
```

Comment:
We might not need to scrape election URLs from ALL network sites to determine if there is an active election, since an automated post is made on per-site metas when an election is announced, we could just use a filtered questions query like this (https://stackexchange.com/filters/421979/all-elections), and use the RSS feed (https://stackexchange.com/feeds/tagsets/421979/all-elections?sort=newest), and filter titles by "community moderator election" without "results"

Reply:
To get election url from meta posts, e.g.:
`https://linguistics.meta.stackexchange.com/questions/2004/2021-community-moderator-election`

Get hostname, drop `meta.`, and append `/election`. 

result: https://linguistics.stackexchange.com/election

The election scraper will then detect the election number (commit https://github.com/samliew/se-electionbot/commit/708948b3e97eeb60a6d8bb05f32e33a7ff767af2)

Reply:
The above RSS feed method doesn't work for the 5 non-english sites, so election pages on those sites have to be scraped separately to detect if they are running one.

Reply:
I was actually going to go all-in on this and add several layers of redundancy in case one fails :) Thus such a bizarre modular PR with a bunch of branches around it. Currently, we have 2 options if I am not mistaken: requesting the API for the list of sites and then parsing network URLs from the resulting array and going for the automation tools provided by SE.

Regarding the former, I want to make it a robust fallback option in case anything goes wrong with faster options. As for the latter - I like the idea of going for the RSS feed to get the URLs as a primary option - I am a little worried about title matching knowing how wobbly SE's approach may be in terms of keeping structure, but this is where the fallback will simply take the lead should anything unexpected happen.